wasi/wasi.go

```diff
@@ -82,9 +84,60 @@ type API interface {
 	// See https://en.wikipedia.org/wiki/Null-terminated_string
 	ArgsSizesGet(ctx wasm.HostFunctionCallContext, resultArgc, resultArgvBufSize uint32) Errno
 
-	// TODO: EnvironGet(ctx api.hostFunctionCallContext, environ, environBuf uint32) Errno
+	// EnvironGet is the WASI function that reads environment variables.
+	//
+	// There are two parameters. Both are offsets in api.HostFunctionCallContext Memory. If either are invalid due to
+	// memory constraints, this returns ErrnoFault.
+	//
+	// * environ - is the offset to begin writing environment variables offsets in uint32 little-endian encoding.
+	//   * EnvironSizesGet result environc * 4 bytes are written to this offset
+	// * environBuf - is the offset to write the null terminated environment variables in the form of "key=val", the same format as os.Environ(), to wasm.MemoryInstance.
```

Comment:
I think we should avoid mentions to `wasm.MemoryInstance` as it's hidden behind the interface

Reply:
That makes sense. `ctx.Memory.Buffer` will need the update as well. So, let's change them to `wasm.Memory`, the interface name.

Reply:
yeah and I will fix any other related drift in next PR. sorry